### PR TITLE
zero-page section flag

### PR DESCRIPTION
**Pull request by laoo**
Pull request covering issue #6 in code-base updated to newest LLVM

### CODE REVIEW

**Comment by johnwbyrd** [comment]
Just wanted to leave some notes on this merge request, just so you know I am not ignoring it.

I am able to follow through your intentions on zeropage.s, but this is not an actual test suite.  The program does run as part of building the check-llvm-mc-mos target, but its output is not checked against a known correct output.  See hello-world.s as an example of how to check output.

A bigger issue, however, is that address width determination does not happen until the relaxation step of the assembler.  In other words, just using the -assemble flag, and not compiling the file to an .obj, does not exercise the logic that you added for setting the zero-page section flag.

In short, the implementation seems superficially correct to me, but the code for testing it 1) doesn't seem to do any actual tests; and 2) doesn't exercise the code you wrote.  

These issues are subtle, and I can write a test suite for your code when I get around to it; or you can look at hello-world.s and get an idea of how a proper test suite can be structured.

Lastly, please don't be discouraged from these comments.  Your implementation is certainly clean, and I would very much like to include it, and it will be included, when either of us can finish the test suite properly for your code.

**Comment by laoo** [comment]
I think that I understand the issue here that I haven't done the proper test. I was mindlessly imitating other tests. The problem I'm facing now is that have no clue where to find check-llvm-mc-mos target or how to create it. Could you point me to some direction?
And I'm not discouraged. I hope that I'll understand the LLVM infrastructure soon and could add some less trivial stuff, but for now I'm just poking it gently to not break anything :)

**Comment by johnwbyrd** [comment]
`cmake --build [your-build-directory] --target check-llvm-mc-mos`

For what it's worth, I wrote a new test, that you can review on my hotfix/laoo-zp-section branch at:

https://github.com/johnwbyrd/llvm-mos/blob/hotfix/laoo-zp-section/llvm/test/MC/MOS/zeropage.s

Interestingly, this test passes fine on Windows and Linux, but the test is throwing an error on Macintosh.  That's why we do cross-platform tests :)

**Comment by johnwbyrd** [comment]
I merged your fixes manually, along with adding test cases, in 979f7da.  Thanks for the excellent ground work!
